backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -82,6 +90,10 @@ public Parcel GetNoTracking(int id)
                 .Include(p => p.Buildings).ThenInclude(b => b.BuildingConstructionType)
                 .Include(p => p.Buildings).ThenInclude(b => b.BuildingPredominateUse)
                 .AsNoTracking().SingleOrDefault(u => u.Id == id);
+
+            var agencies = this.User.GetAgencies();
+            parcel.Buildings.RemoveAll(b => b.IsSensitive && !agencies.Contains(b.AgencyId));
+            return parcel;
```

Comment:
Please help me understand how the above mutates the `parcel.Buildings` Enumerable

Reply:
If I follow your code correctly, the "RemoveAll" function takes an Enumerable source: `parcel.Buildings` and return a copy in List form: `items.ToList()`. It will then mutate the list but I don't think that also mutates the underlying Enumerable, does it?

Reply:
I think I've mistakenly mutated the list rather than the ienumerable... good catch